### ISSUE TITLE
CORS allow multiple domains

### ISSUE THREAD

**Issue by marcderhammer** [parse-community/parse-server#8516]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

The current configuration seems to only support one origin or ALL origins (*)

### Feature / Enhancement Description

Modify the middlewares' [allowCrossDomain](https://github.com/parse-community/parse-server/blob/177891ea0eed54eff996d81749cfbfb32156ae87/src/middlewares.js#L380) function to support comma delimited `ALLOW_ORIGIN`, look for a matching origin to the request's origin header and set the response header to the match, otherwise just set the header to the first value in the list of allowed origins

### Example Use Case

```
new ParseServer({
  // rest of config
  allowOrigin: 'https://a.com,https://b.com,https://c.com'
})
```

Then, webapps at `a.com`, `b.com`, and `c.com` will not have CORS issues sending requests, but `d.com` will not be allowed.

### Alternatives / Workarounds

Setting to * (or leaving undefined), allows any origin, but this has security implications.

### 3rd Party References

https://github.com/expressjs/cors

### Notes
I have a patch which I've been using.  I'll make a PR soon

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.1.0-alpha.8](https://github.com/parse-community/parse-server/releases/tag/6.1.0-alpha.8)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.1.0-beta.2](https://github.com/parse-community/parse-server/releases/tag/6.1.0-beta.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.1.0](https://github.com/parse-community/parse-server/releases/tag/6.1.0)
